frontend/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,70 @@
+import { cloneElement } from 'react'
+import {
+  Checkbox as ChakraCheckbox,
+  CheckboxGroup,
+  CheckboxGroupProps as ChakraCheckboxProps,
+  Flex,
+  Input,
+  StylesProvider,
+  useMultiStyleConfig,
+  useStyles,
+  VStack,
+} from '@chakra-ui/react'
+
+export interface CheckboxProps extends ChakraCheckboxProps {
+  options?: string[]
+  other: boolean
+  otherComponent?: JSX.Element // TODO: change type to disjunction of acceptable components
+}
+
+const defaultOtherComponent = <Input placeholder="Please specify"></Input> // TODO: replace with custom input component
+
+const CheckboxOption = ({ option }: { option: string }): JSX.Element => {
+  const styles = useStyles()
+  return (
+    <ChakraCheckbox value={option} __css={styles.container}>
+      {option}
+    </ChakraCheckbox>
+  )
+}
+
+const OtherOption = ({
+  component,
+}: {
+  component: JSX.Element
+}): JSX.Element => {
+  const styles = useStyles()
+  return (
+    <Flex direction="column">
+      <CheckboxOption option="Other" />
+      <Flex pl="48px" mt="2px">
+        {cloneElement(component, {
```

Comment:
I kept the clone element to pass the `onChange` handler to the child

Reply:
You control the implementation, so you can decide to pass any onChange handlers to the children input. This may actually be more complex... should we just stick to only allowing an input field if the others option is enabled? this way the component can be self contained

Reply:
Yeah I think that sounds good haha will do that instead